shell/platform/darwin/macos/framework/Source/FlutterEngine.mm

```diff
@@ -447,9 +447,15 @@ - (void)sendOnChannel:(NSString*)channel
   }
 }
 
-- (void)setMessageHandlerOnChannel:(nonnull NSString*)channel
-              binaryMessageHandler:(nullable FlutterBinaryMessageHandler)handler {
+- (FlutterBinaryMessengerConnection)setMessageHandlerOnChannel:(nonnull NSString*)channel
+                                          binaryMessageHandler:
+                                              (nullable FlutterBinaryMessageHandler)handler {
   _messageHandlers[channel] = [handler copy];
+  return 0;
+}
+
+- (void)cleanupConnection:(FlutterBinaryMessengerConnection)connection {
+  // There hasn't been a need to implement this yet for macOS.
```

Comment:
Should this have a bug/TODO? @stuartmorgan

Reply:
The reason I had to implement this was because of the accessibility bridge on iOS, so it wasn't needed for macOS.  This problem I suspect may show up in other usage of channels though.  Channels behave the same way they did previously without implementing this.  I don't think it would hurt to implement it though, just hasn't proved itself necessary.  The  connection collection was pulled out to help share code with macos if we need it.